sycl/doc/extensions/experimental/sycl_ext_codeplay_kernel_fusion.asciidoc

```diff
@@ -0,0 +1,678 @@
+
+|`bool is_in_fusion_mode() const`
+|Returns true if the wrapped SYCL `queue` is currently in fusion mode.
+
+|===
+
+==== Properties
+
+Next to the new API functions and classes described above, this extension also
+adds new properties that are described in Table 3.
+
+Table 3. New properties for kernel fusion.
+
+|===
+|Property|Description
+
+|`sycl::ext::codeplay::property::queue::enable_fusion`
+|This property enables kernel fusion for the queue. If a `fusion_wrapper` object
+is constructed on a queue without this property, an `exception` with `errc::invalid`
+error code is thrown.
+
+|`sycl::ext::codeplay::property::no_barriers`
+
+|If the property list passed to `fusion_wrapper::complete_fusion()` contains this
+property, no barriers are introduced between kernels in the fused kernel.
+
+|`sycl::ext::codeplay::property::promote_local`
+|This property gives a hint to the JIT compiler to try to internalize a given
+ argument via promotion to local memory (see local and private internalization
+ in <<internalization>>).
+
+This property can be passed to the `accessor` constructor, giving a more
+ granular control, or to the `buffer` constructor, in which case all the
+ `accessors` will inherit this property (unless overridden).
+
+|`sycl::ext::codeplay::property::promote_private`
+|This property gives a hint to the JIT compiler to try to internalize a given
+ argument via promotion to private memory (see local and private internalization
+ in <<internalization>>).
+
+This property can be passed to the `accessor` constructor, giving a more
+ granular control, or to the `buffer` constructor, in which case all the
+ `accessors` will inherit this property (unless overridden).
+
+|`sycl::ext::codeplay::property::force_fusion`
+
+|This property forces the SYCL runtime implementation to perform fusion if it is
+possible to do so. Implementations must not defer kernel fusion, even if they
+deemed the fusion to be non-profitable, e.g., based on some profitability
+analysis.
+
+This property can be passed to `fusion_wrapper::complete_fusion()`.
+
+|===
+
+==== New Queue Member Functions
+
+To support querying if a `queue` can be used for fusion, i.e., can be wrapped by
+a `fusion_wrapper` object, this extension adds a new member function to the
+`queue` class. 
+
+Table 4. Added member functions of the `queue` class
+
+|===
+|Member Function|Description
+
+|`bool queue::ext_codeplay_supports_fusion() const`
+
+|Returns true if the SYCL `queue` was created with the `enable_fusion` property.
+Equivalent to `has_property<ext::codeplay::property::queue::enable_fusion>()`.
+
+|===
+
+==== Additional Device Information Descriptors
+
+To support querying whether a SYCL device and the underlying platform support
+kernel fusion before constructing a queue with property
+`ext::codeplay::property::queue::enable_fusion`, the following device
+information descriptor is added as part of this extension proposal. 
+
+Table 5. Added device information descriptors
```

Comment:
These should also be in the experimental namespace `sycl::ext::codeplay::experimental::info::device`.

Reply:
The forms of the info descriptors with the leading "ext_intel" are deprecated.  The preferred form is to use the namespace, similar to what you did with the properties.  See here for an example:

https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_device_info.md#deprecated-queries

Newly added info descriptors should not follow the deprecated form.  Just define the descriptor in the namespace.

Reply:
I wasn't aware this naming scheme was deprecated, sorry. The device information descriptor was updated in the most recent commit.